### ISSUE TITLE
Category indicator not displaying correctly

### ISSUE THREAD

**Issue by huynguyen93** [nolanlawson/emoji-picker-element#206]
Hello,

First of all, thanks for your work with this project, it's great!

I just found a problem with the category indicator, it doesn't display in 2 cases:
- when the container size is smaller than 400px
- when the document direction is RTL (ie: when language is Hebrew)

I attach 2 short videos so it's easier to see

https://user-images.githubusercontent.com/16201014/128686932-bc5b0c33-d3a3-4e0b-86e3-0589378ec083.mp4

https://user-images.githubusercontent.com/16201014/128686936-aa6b039a-7a1f-4a23-acca-23e129848b63.mp4

I'm sorry, I don't have enough time to look into the code to suggest a fix...

**Comment by huynguyen93** [comment]
From what I checked quickly, maybe we should do CSS border bottom for the category button instead of manual calculation?

**Comment by nolanlawson** [comment]
Thanks for reporting. For the RTL issue, I think it might be a separate issue, but for the "too small" issue, I had kind of assumed that the picker would never need to be scrolled horizontally like that. Although I suppose there are some valid use cases for that (small feature phones?).

The manual calculation is done for performance reasons, but maybe Chrome no longer has the perf issue described here so we can roll it back. I'll look into it.

https://github.com/nolanlawson/emoji-picker-element/blob/13e6bcb068bb126c6a69eae41b7bae32092d391b/src/picker/components/Picker/Picker.js#L261-L272

**Comment by huynguyen93** [comment]
Yep, I was testing with an iphone 6s, which has the width of only 320px including some padding/margin (I'm making a widget that can be used in different places).
My current workaround is to make it min-width 400px and show a horizontal scroll but I guess it's better to have a more responsive display...

Thanks for looking into it!

**Comment by nolanlawson** [comment]
The solution I intended for small screens was to change `--num-columns` and/or `--category-emoji-size`. E.g. you could do something like:

```css
@media screen and (max-width: 375px) {
  emoji-picker {
    --num-columns: 6;
    --category-emoji-size: 1.125rem;
  }
}
```

I didn't really intend for it to ever be horizontally scrollable, but I guess I didn't do a good job explaining that in the docs. :sweat_smile: Or maybe I should just include some built-in CSS for small screens.

**Comment by nolanlawson** [comment]
OK, looks like the manual calculation (exact pixels vs percent-based) has nothing to do with this. There are two separate bugs:

- when horizontal scrollbar is visible, indicator is wrong
- when in RTL mode, indicator is wrong

BTW the reason I don't use a border-bottom is because there's no way to animate it performantly (GPU-accelerated).

**Comment by nolanlawson** [comment]
Opened a new issue for RTL: https://github.com/nolanlawson/emoji-picker-element/issues/212

**Comment by huynguyen93** [comment]
Thanks, just my thought, we don't really need the "slide" animation, a simple border bottom works well IMO (it clearly indicates that we're in that category, and doesn't look too bad)

**Comment by nolanlawson** [comment]
I added some docs around this: https://github.com/nolanlawson/emoji-picker-element/pull/215

I don't plan to switch to `border-bottom` because I like the slide animation. Also I don't believe the categories should be horizontally scrollable on mobile – I think it's poor design because it reduces the discoverability (especially with hidden scrollbars, the user may not realize it can scroll).

I would prefer for people to add their own CSS so that the categories are not horizontally scrollable, but in theory it should be possible to make the indicator adapt to the width of the categories, so I'll leave this bug open.
